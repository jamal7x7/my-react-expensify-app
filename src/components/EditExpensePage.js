import React from 'react'

const EditExpensePage = () => (
  <div>
    Hello from edit expense component!
  </div>
)

export default EditExpensePage
