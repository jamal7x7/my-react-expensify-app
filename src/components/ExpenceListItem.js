import React from 'react'
import { connect } from 'react-redux'
import { removeExpence } from '../actions/expences'
import configureStore from '../store/configureStore'


export const ExpenceDiscription = ({ dispatch, id, discription, amount, createdAt }) => (
  <div>

    <h3 > { discription } </h3>
    <p> {amount} - {createdAt} </p>
    <button
      onClick={e => {
        dispatch(removeExpence({id}))
      }}
      >Remove
    </button>
  </div>
)

const mapStateToProps = state => {
  return {
    expences: state.expences
  }
}

export default connect( mapStateToProps )(ExpenceDiscription)
