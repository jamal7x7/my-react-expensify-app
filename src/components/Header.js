import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => (
  <header >
    <h1 className='title' >ExpensifyÂ°</h1>
    <div className='navWrapper' >
      <div className='nav'>
        <NavLink className='navLink' activeClassName='is-active' to="/" exact={true}>  Dashboard  </NavLink>
        <NavLink className='navLink' activeClassName='is-active' to="/create">  Create  </NavLink>
        <NavLink className='navLink' activeClassName='is-active' to="/edit">  Edit  </NavLink>
        <NavLink className='navLink' activeClassName='is-active' to="/help">  Help  </NavLink>
      </div>
    </div>

  </header >
)

export default Header
