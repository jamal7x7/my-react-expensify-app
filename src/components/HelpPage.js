import React from 'react'

const HelpPage = () => (
  <div>
    Hello from help expense component!
  </div>
)

export default HelpPage
