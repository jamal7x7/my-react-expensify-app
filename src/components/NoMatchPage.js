import React from 'react'
import {Link} from 'react-router-dom'

export const NoMatchPage = () => (
  <div>
    <h1>404!!!</h1>
    <h3>Page not found!</h3>
    <Link to="/">Go Home</Link>

  </div>
)

export default NoMatchPage
