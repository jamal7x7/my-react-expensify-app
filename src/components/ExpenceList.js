import React from 'react'
import { connect } from 'react-redux'
import ExpenceListItem from './ExpenceListItem'
import selectExpence from '../selectors/expences'


export const ExpenceList = (props) => (
  <div>
    <h1>Expence List</h1>
    {
      props.expences.length !== 0 ? (
        props.expences.map( expence => {
          return <ExpenceListItem key={expence.id} {...expence} />
        })
      ) : (
        <p>No expences</p>
      )
    }
  </div>
)

const mapStateToProps = state => {
  return {
    expences: selectExpence(state.expences, state.filters)
  }
}

export default connect( mapStateToProps )(ExpenceList)
