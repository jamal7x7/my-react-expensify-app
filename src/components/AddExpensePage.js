import React from 'react'
import { connect } from 'react-redux'
import { addExpence } from '../actions/expences'
import ExpenceList from './ExpenceList'

const AddExpensePage = (props) => (
  <div>
    Hello from add expense component!<br/><br/><br/>

    <form action="" value="" onSubmit= { e => {
      e.preventDefault()
      e.target.elements.discription.value && props.dispatch(addExpence({
        discription: e.target.elements.discription.value,
        amount: e.target.elements.amount.value || 'not specified',
        createdAt: e.target.elements.createdAt.value || 'not specified'
      }))
      e.target.elements.discription.value=''
      e.target.elements.amount.value=''
      e.target.elements.createdAt.value=''
    }} >
      Discription : <input autoFocus type="text" name="discription" value={props.expences.discription} /> <br/><br/>
      Amount ($): <input type="number" name="amount" value={props.expences.amount }/> <br/><br/>
      Created at : <input type="text" name="createdAt" value={props.expences.createdAt} /> <br/><br/><br/>
      <button> ADD</button>
      <br/><br/><br/>
      <ExpenceList />
    </form>

  </div>
)

const mapStateToProps = state => (
  {
    expences: state.expences
  }
)

export default connect (mapStateToProps) (AddExpensePage)
