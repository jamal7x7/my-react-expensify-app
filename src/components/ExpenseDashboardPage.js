import React from 'react'
import ExpenceList from './ExpenceList'
import ExpenceListFilters from './ExpenceListFilters'

export const ExpenseDashboardPage = () => (
  <div>
    <ExpenceListFilters />
    <ExpenceList />
  </div>
)

export default ExpenseDashboardPage
