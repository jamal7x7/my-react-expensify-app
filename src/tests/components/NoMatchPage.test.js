import React from 'react'
import { shallow } from 'enzyme'
import { NoMatchPage } from '../../components/NoMatchPage'


test('should render NoMatchPage correctly', () => {
  const wrapper = shallow(<NoMatchPage />)
  expect(wrapper).toMatchSnapshot()
})
