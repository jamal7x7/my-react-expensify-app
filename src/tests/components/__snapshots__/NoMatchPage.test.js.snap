// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NoMatchPage correctly 1`] = `
<div>
  <h1>
    404!!!
  </h1>
  <h3>
    Page not found!
  </h3>
  <Link
    replace={false}
    to="/"
  >
    Go Home
  </Link>
</div>
`;
